h2/src/main/org/h2/tools/Recover.java

```diff
@@ -1031,8 +1031,6 @@ private void dumpPageLogStream(PrintWriter writer, int logKey,
                 int sessionId = in.readVarInt();
                 String transaction = in.readString();
                 writer.println("-- prepare commit " + sessionId + " " + transaction);
-            } else if (x == PageLog.NOOP) {
-                // nothing to do
             } else if (x == PageLog.CHECKPOINT) {
```

Comment:
Also wrong.

At this point I stopped checking - please don't just routinely accept warnings from such tools. You need to carefully analyse the surrounding code and then determine if the warning is valid - such tools often produce false positives.

Reply:
>Also wrong.

Can you please explain why?
As I can see `x` is already compared with `PageLog.NOOP` in `if-else-if` chain. See - https://github.com/h2database/h2database/blob/master/h2/src/main/org/h2/tools/Recover.java#L899
So it doesn't make sense to check it again.

Reply:
Yeah, you are right